frontend/src/store/eam-worker-configurate/actions.ts

```diff
@@ -1,34 +1,41 @@
 import { createAsyncThunk } from '@reduxjs/toolkit';
 import {
   EAMWorkerCreateRequestDto,
-  EAMWorkerCreateResponseDto,
   AsyncThunkConfig,
   EAMGroupGetByTenantRequestParamsDto,
   EAMGroupGetByTenantResponseDto,
 } from 'common/types/types';
 import { ActionType } from './common';
 import { getRandomId } from 'helpers/helpers';
-import { AppRoute } from 'common/enums/enums';
+import { CSV } from 'common/enums/enums';
 
 const workerCreate = createAsyncThunk<
-  EAMWorkerCreateResponseDto,
+  File,
   EAMWorkerCreateRequestDto,
   AsyncThunkConfig
 >(
   ActionType.CREATE_WORKER,
   async (payload: EAMWorkerCreateRequestDto, { extra }) => {
-    const { workerApi, navigation, notification } = extra;
+    const { workerApi, notification, saver } = extra;
 
-    const worker = await workerApi.createWorker({
+    const password = getRandomId();
+
+    await workerApi.createWorker({
       ...payload,
-      password: getRandomId(),
+      password,
     });
 
-    navigation.push(AppRoute.EAM);
+    const csvFile = saver.saveCSV(
+      [
+        [CSV.NAME, payload.name],
+        [CSV.PASSWORD, password],
+      ],
+      `${payload.name}-worker-credentials`,
+    );
```

Comment:
on this step, you need to store columns in the storage, and then by clicking on the `download creds button`, you need to call another action that will trigger saving